File: AAC-core-demo/savedstate/savedstate/src/main/java/androidx/savedstate/ViewTreeSavedStateRegistryOwner.java
package androidx.savedstate;

import android.view.View;

import org.jetbrains.annotations.NotNull;

import androidx.annotation.Nullable;
import kotlin.jvm.JvmName;
import kotlin.jvm.internal.Intrinsics;
import kotlin.sequences.SequencesKt;

public final class ViewTreeSavedStateRegistryOwner {
    @JvmName(
            name = "set"
    )
    public static final void set(@NotNull View $this$setViewTreeSavedStateRegistryOwner, @Nullable SavedStateRegistryOwner owner) {
        Intrinsics.checkNotNullParameter($this$setViewTreeSavedStateRegistryOwner, "<this>");
        $this$setViewTreeSavedStateRegistryOwner.setTag(R.id.view_tree_saved_state_registry_owner, owner);
    }

    @JvmName(
            name = "get"
    )
    @Nullable
    public static final SavedStateRegistryOwner get(@NotNull View $this$findViewTreeSavedStateRegistryOwner) {
        Intrinsics.checkNotNullParameter($this$findViewTreeSavedStateRegistryOwner, "<this>");
        return (SavedStateRegistryOwner) SequencesKt.firstOrNull(
                SequencesKt.mapNotNull(
                        SequencesKt.generateSequence($this$findViewTreeSavedStateRegistryOwner, null), null
                )
        );
    }
}

File: AAC-core-demo/savedstate/savedstate/src/main/java/androidx/savedstate/ViewTreeSavedStateRegistryOwner.kt
package androidx.savedstate

import android.view.View

/**
 * Set the [SavedStateRegistryOwner] responsible for managing the saved state for this [View]
 * Calls to [get] from this view or descendants will return `owner`.
 *
 * This is is automatically set for you in the common cases of using fragments or
 * ComponentActivity.
 *
 *
 * This should only be called by constructs such as activities or fragments that manage
 * a view tree and their saved state through a [SavedStateRegistryOwner]. Callers
 * should only set a [SavedStateRegistryOwner] that will be *stable.* The
 * associated [SavedStateRegistry] should be cleared if the view tree is removed and is
 * not guaranteed to later become reattached to a window.
 *
 * @param owner The [SavedStateRegistryOwner] responsible for managing the
 * saved state for the given view
 */
@JvmName("set")
fun View.setViewTreeSavedStateRegistryOwner(owner: SavedStateRegistryOwner?) {
    setTag(R.id.view_tree_saved_state_registry_owner, owner)
}

/**
 * Retrieve the [SavedStateRegistryOwner] responsible for managing the saved state
 * for this [View].
 * This may be used to save or restore the state associated with the view.
 *
 * The returned [SavedStateRegistryOwner] is managing all the Views within the Fragment
 * or Activity this [View] is added to.

 * @return The [SavedStateRegistryOwner] responsible for managing the saved state for
 * this view and/or some subset of its ancestors
 */
@JvmName("get")
fun View.findViewTreeSavedStateRegistryOwner(): SavedStateRegistryOwner? {
    return generateSequence(this) { view ->
        view.parent as? View
    }.mapNotNull { view ->
        view.getTag(R.id.view_tree_saved_state_registry_owner) as? SavedStateRegistryOwner
    }.firstOrNull()
}